File: src/main/java/com/cfgglobal/test/dao/RuleDao.java
package com.cfgglobal.test.dao;

import com.cfgglobal.test.dao.base.BaseDao;
import com.cfgglobal.test.domain.Rule;
import io.vavr.collection.List;
import io.vavr.control.Option;
import org.springframework.stereotype.Repository;

@Repository
public interface RuleDao extends BaseDao<Rule, Long> {

    Option<Rule> findByName(String name);

    List<Rule> findByType(String basic);
}

File: src/main/kotlin/com/cfgglobal/test/dao/RuleDao.kt
package com.cfgglobal.test.dao

import com.cfgglobal.test.dao.base.BaseDao
import com.cfgglobal.test.domain.Rule
import org.springframework.stereotype.Repository

@Repository
interface RuleDao : BaseDao<Rule, Long> {

    fun findByName(name: String): Rule

    fun findByType(basic: String): List<Rule>
}